### PR TITLE
DM-36604: Create rubin-env 5.0.0 and rubin-env-developer.

### PR DESCRIPTION
**Pull request by ktlim**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by ktlim** [comment]
@conda-forge-admin, please rerender

**Comment by ktlim** [comment]
@conda-forge-admin, please rerender

**Comment by ktlim** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml [hunk 1]
```diff
@@ -92,7 +92,7 @@ outputs:
         - eigen {{ eigen }}.*
         - eups
         - firefly-client
-        - galsim <2.4.0
+        - galsim
```
Comment (beckermr) [inline recipe/meta.yaml:95]: You want a lower bound here since old galsim now won't work with the new Rubin code.

Comment (beckermr) [reply recipe/meta.yaml:95]: Ah ok. If that's true then ignore me.

Comment (ktlim) [reply recipe/meta.yaml:95]: Yes, we do support both.

File: recipe/meta.yaml [hunk 2]
```diff
@@ -141,44 +139,41 @@ outputs:
...
         - schwimmbad
         - scikit-image !=0.18
         - scikit-learn
-        # scipy 1.9 breaks a meas_base test; see DM-35777
-        - scipy <1.9
+        - scipy
         - seaborn
         - skyproj
-        - spherematch <0.10
+        - spherematch
```
[the remark this answers is not in the document]
Comment (ktlim) [reply recipe/meta.yaml:176]: https://jira.lsstcorp.org/browse/DM-35083 says:
"Second, updates will be made to isolatedStarAssociation.py that will be able to handle both versions of spherematch."

File: recipe/meta.yaml (hunk 1 above)
Comment (ktlim) [reply recipe/meta.yaml:95]: https://github.com/lsst/meas_extensions_shapeHSM/pull/39/files

File: recipe/meta.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (beckermr) [reply recipe/meta.yaml:176]: Ahhhhh nvm then.

**Review by beckermr — APPROVED** [review]

**Review by erykoff — APPROVED** [review]

**Pull request merged**